### PR TITLE
adding huggingface inference support + litellm debugger

### PR DESCRIPTION
**Pull request by krrishdholakia**
Hi @mrT23 @okotek,

Following up on @dzlab's request and adding Huggingface Inference support (we also support palm - via vertex ai, together_ai + baseten deployments, can add those too). 

We also introduced a free debugging tool - added instructions on how to use that in the INSTALL.MD docs. Let me know if another place might work better - https://docs.litellm.ai/docs/debugging/hosted_debugging. 

Can add additional docs + testing if the initial PR looks good.

### CODE REVIEW

**Comment by mrT23** [comment]
Thanks a lot @krrishdholakia,

Can you elaborate a bit more on what it means to support "HUGGINGFACE" ?
do you have an easy example to demonstrate the usage of a HUGGINGFACE model ?

File: pr_agent/algo/ai_handler.py [hunk 1]
```diff
@@ -43,6 +44,12 @@ def __init__(self):
                 litellm.cohere_key = get_settings().cohere.key
             if get_settings().get("REPLICATE.KEY", None):
                 litellm.replicate_key = get_settings().replicate.key
+            if get_settings().get("REPLICATE.KEY", None):
+                litellm.replicate_key = get_settings().replicate.key
+            if get_settings().get("HUGGINGFACE.KEY", None):
+                litellm.huggingface_key = get_settings().huggingface.key
+            if get_settings().get("HUGGINGFACE.KEY", None):
+                litellm.huggingface_key = get_settings().huggingface.key
```
Comment (dzlab) [inline pr_agent/algo/ai_handler.py:52]: this looks like duplication, no?

**Comment by dzlab** [comment]
@krrishdholakia how does this debugger tool compares to [langsmith](https://docs.smith.langchain.com/)?

**Comment by krrishdholakia** [comment]
@dzlab haven't tried langsmith out - it looks great! I just needed a way to tell if the raw api calls were being made. 

@mrT23 It works with the HF Inference API - https://docs.litellm.ai/docs/completion/supported#hugging-face-inference-api

We also support deployment providers like Baseten, TogetherAI. 

I believe Baseten has a deployment of WizardLM you can use out of the box - https://docs.litellm.ai/docs/completion/supported#baseten-models

**Review by mrT23 — APPROVED** [review]

**Comment by mrT23** [comment]
ok. i will merge and try the example you sent :)

HF Inference API is indeed a popular service

**Comment by mrT23** [comment]
Preparing review...

File: requirements.txt [hunk 2]
```diff
@@ -14,4 +14,4 @@ GitPython~=3.1.32
 litellm~=0.1.351
 PyYAML==6.0
 starlette-context==0.3.6
-litellm~=0.1.351
+litellm~=0.1.445
```
Comment (mrT23) [inline requirements.txt:17]: **Suggestion:** The 'litellm' package is listed twice with different versions. Please ensure to keep only the required version.
```suggestion
litellm~=0.1.445
```

File: pr_agent/algo/ai_handler.py (hunk 1 above)
Comment (mrT23) [inline pr_agent/algo/ai_handler.py:52]: **Suggestion:** There seems to be a duplication of code for setting the 'replicate_key' and 'huggingface_key'. Please ensure that each key is set only once.
```suggestion
            if get_settings().get('REPLICATE.KEY', None):
                litellm.replicate_key = get_settings().replicate.key
            if get_settings().get('HUGGINGFACE.KEY', None):
                litellm.huggingface_key = get_settings().huggingface.key
```

Comment (mrT23) [reply pr_agent/algo/ai_handler.py:52]: @dzlab  "/improve" agrees with you :-)

File: requirements.txt (hunk 2 above)
Comment (okotek) [inline requirements.txt:17]: Please also change in pyproject.toml

**Review by okotek — CHANGES_REQUESTED** [review]

**Review by mrT23 — APPROVED** [review]

**Comment by mrT23** [comment]
i fixed the minor issues in another commit

**Comment by krrishdholakia** [comment]
Thanks! @mrT23

**Pull request merged**